src/main/java/Elevator.java

```diff
@@ -0,0 +1,141 @@
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class Elevator implements RandomFloorsAndPassengers {
+    private static int MAX_SEATS_IN_ELEVATOR = 5;
+    private final Building building;
+    private Queue<Passenger> queueInElevator;
+    private int passengersInElevator;
+
+    public Elevator() {
+        queueInElevator = new LinkedList<>();
+        passengersInElevator = 0;
+        building = new Building();
+    }
+
+    public Queue<Passenger> getQueueInElevator() {
+        return queueInElevator;
+    }
+
+    public void setQueueInElevator(Queue<Passenger> queueInElevator) {
+        this.queueInElevator = queueInElevator;
+    }
+
+    public int getPassengersInElevator() {
+        return passengersInElevator;
+    }
+
+    public void setPassengersInElevator(int passengersInElevator) {
+        this.passengersInElevator = passengersInElevator;
+    }
+
+    public int getHighestFloorWherePassengerWantsRise(Queue<Passenger> passengerQueue) {
+        int retFloor = -1;
+        for (Passenger passengers : passengerQueue) {
+            if (passengers.getFloor() > retFloor) {
+                retFloor = passengers.getFloor();
+            }
+        }
+        return retFloor;
+    }
+
+    public void loadElevator(int floor) {
+        Queue<Passenger> passengerQueue = building.getArrayOfFloors()[floor].getQueue();
+        while (getPassengersInElevator() != MAX_SEATS_IN_ELEVATOR
+                && passengerQueue.peek() != null) {
+            queueInElevator.add(passengerQueue.peek());
+            passengerQueue.poll();
+            building.getArrayOfFloors()[floor].setQueue(passengerQueue);
+            setPassengersInElevator(queueInElevator.size());
+            building.getArrayOfFloors()[floor].setPassengersOnFloor(building
+                    .getArrayOfFloors()[floor].getPassengersOnFloor() - 1);
+        }
+    }
+
+    public int unloadElevator(int floor) {
+        Queue<Passenger> passengerQueue = getQueueInElevator();
+        int retValue = 0;
+        int counterIteration = 0;
+        while (counterIteration != MAX_SEATS_IN_ELEVATOR && passengerQueue.peek() != null) {
+            if (passengerQueue.peek().getFloor() == floor) {
+                passengerQueue.poll();
+                setQueueInElevator(passengerQueue);
+                setPassengersInElevator(passengerQueue.size());
+                retValue++;
+            } else {
+                passengerQueue.add(passengerQueue.peek());
+                passengerQueue.poll();
```

Comment:
```suggestion
                passengerQueue.add(passengerQueue.poll());
```
can we use poll right away